File: Difficulty: Easy/Array Leaders/array-leaders.js
class Solution {
    // Function to find the leaders in the array.
    leaders(a) {
        let ans = [];
        let n = a.length;

        ans.push(a[n - 1]); // Use push() in JavaScript
        let max = a[n - 1];

        for (let i = n - 2; i >= 0; i--) {
            if (a[i] >= max) {
                max = a[i];
                ans.unshift(a[i]); // unshift() adds to the beginning
            }
        }

        return ans;
    }
}

File: Difficulty: Easy/Array Leaders/array-leaders.java
class Solution {
    static ArrayList<Integer> leaders(int arr[]) {
        // code here
        int n = arr.length;
        int max = arr[n-1];
        ArrayList<Integer> ans = new ArrayList<>();
        ans.add(arr[n-1]);
        
        for(int i = n-2; i>=0; i--){
            if(arr[i] >= max){
                max = arr[i];
                ans.add(max);
            }
        }
        
        Collections.reverse(ans);
        return ans;
    }
}
